clicommand/agent_start.go

```diff
@@ -511,9 +514,20 @@ var AgentStartCommand = cli.Command{
 		},
 		cli.BoolFlag{
 			Name:   "no-command-eval",
-			Usage:  "Don't allow this agent to run arbitrary console commands, including plugins",
+			Usage:  "Don't allow this agent to run arbitrary console commands; also disables plugins by default - can only be used with command-mode 'legacy'",
 			EnvVar: "BUILDKITE_NO_COMMAND_EVAL",
 		},
+		cli.StringFlag{
+			Name:   "command-mode",
+			Value:  job.CommandModeLegacy,
+			Usage:  "Specifies how the agent should run commands - can be 'legacy', 'shell' or 'program'",
+			EnvVar: "BUILDKITE_COMMAND_MODE",
+		},
+		cli.BoolTFlag{
+			Name:   "command-repo-only",
+			Usage:  "Command must refer to a program in the repo - only used if 'command-mode' is 'program'",
```

Comment:
```suggestion
			Usage:  "Command must refer to a program in the repo - can only used be set if 'command-mode' is 'program'. Note: this won't prevent you from executing programs from outside the repo by invoking them from scripts inside the repo.",
```

I think it's important to add this disclaimer, even if the original wording did not guarantee the opposite of it.